node/src/components/storage.rs

```diff
@@ -473,8 +478,46 @@ impl Storage {
             fault_tolerance_fraction,
             max_ttl,
             metrics,
+            force_resync_pid_file: None,
         };
 
+        if force_resync {
+            // Check if resync is already in progress. Force resync will kick
+            // in only when the pid file was cleanly initialized and acquired
+            // (i.e. it didn't exist before).
+            match PidFile::acquire(component.root_path().join(FORCE_RESYNC_PID_FILE_NAME)) {
+                utils::pid_file::PidFileOutcome::Clean(pid_file) => {
+                    // When no pid file was detected, initialize force resync.
+                    info!("Initializing force resync.");
+                    // Default `storage.completed_blocks`.
+                    component.completed_blocks = Default::default();
+                    component.persist_completed_blocks()?;
```

Comment:
Is `PidFile` able to clean itself up? Or shall we delete the `pid_file` here to allow retry on next node startup?

Reply:
This `PidFile` is not supposed to clean itself up, it should be persistent. Intended use case is that operators will delete this file if they want to enable the `force-resync` behavior repeatedly.